packages/generator/src/generator.js

```diff
@@ -273,6 +320,19 @@ export default class Generator {
     return true
   }
 
+  async addPayloadScript (route, html) {
+    const windowNamespace = this.options.globals.context(this.options.globalName)
+    const chunks = html.split(`<script>window.${windowNamespace}=`)
+    const [pre] = chunks
+    const payload = chunks[1].split('</script>').shift()
+    const post = chunks[1].split('</script>').slice(1).join('</script>')
+
+    // Write payload.js file
+    await fsExtra.writeFile(path.join(this.payloadDir, route, 'payload.js'), `window.${windowNamespace}=${payload}`, 'utf-8')
```

Comment:
Have you thought about a caching solution (e.g. my proposed build hash option or sth. more customizable) so we can ensure payloads can be fully cached? 😋

Reply:
Well actually I want to separate the **build** and the **generation** so if only the content changes, only the payloads will change and no webpack build is necessary.

Also, since they live inside `dist/` they will be cached on any static hosting.

Reply:
> Also, since they live inside dist/ they will be cached on any static hosting.

Without fingerprinting/hashes, this could easily cause caching issues when you treat them as immutable (as most of the generated JS files should be treated) ☺️

Reply:
Actually the payloadPath includes an exportTime so the path will be unique for each « nuxt export » call 😁

Reply:
Ohh, must've missed that. Clever 👍🏻